spec/lib/decidim/duplicates_metadata_spec.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+require "decidim/duplicates_metadata"
+
+describe Decidim::DuplicatesMetadata do
+  subject { described_class.new }
+
+  let(:organization) { create(:organization) }
+
+  let!(:user) { create(:user, organization: organization) }
+  let!(:user2) { create(:user, organization: organization) }
+
+  let!(:socio_authorization) do
+    create(:authorization,
+           user: user,
+           metadata: { "email" => user.email, "postal_code" => "12345", "city" => "Barcelona", "phone_number" => "01234567889" },
+           name: "extended_socio_demographic_authorization_handler")
+  end
+  let!(:socio_authorization_user2) do
+    create(:authorization,
+           user: user2,
+           metadata: { "email" => user2.email, "postal_code" => "54321", "city" => "Paris", "phone_number" => "9876543210" },
+           name: "extended_socio_demographic_authorization_handler")
+  end
+
+  let!(:data_authorization) do
+    create(:authorization,
+           user: user,
+           metadata: { "minmum_age" => true, "gdpr" => true, "zipcode" => "12345" },
+           name: "data_authorization_handler")
+  end
+  let!(:data_authorization_user2) do
+    create(:authorization,
+           user: user2,
+           metadata: { "minmum_age" => true, "gdpr" => true, "zipcode" => "54321" },
+           name: "data_authorization_handler")
+  end
+
+  context "when user has no metadata" do
```

Comment:
It is easier if we use `shared_examples`  and `it_behaves_like` clausure to simplify reading and file